Controller/SubscriptionWallController.php

```diff
@@ -0,0 +1,175 @@
+ */
+class SubscriptionWallController
+{
+    /**
+     * @var TwigEngine
+     */
+    protected $templating;
+
+    /**
+     * @var EntityManager
+     */
+    protected $em;
+
+    /**
+     * @var ChannelManager
+     */
+    protected $cm;
+    /**
+     * @var Request
+     */
+    protected $request;
+
+    /**
+     * @var FormFactory
+     */
+    protected $form;
+    /**
+     * @var RouterInterface
+     */
+    protected $router;
+
+    /**
+     * @var FlashMessage
+     */
+    protected $flashMessage;
+
+    /**
+     * @param TwigEngine $templating
+     */
+    public function __construct(TwigEngine $templating, EntityManager $em, ChannelManager $cm, FormFactory $form, RouterInterface $router, RequestStack $requestStack, FlashMessage $flashMessage)
+    {
+        $this->templating = $templating;
+        $this->em = $em;
+        $this->cm = $cm;
+        $this->request = $requestStack->getCurrentRequest();
+        $this->form = $form;
+        $this->router = $router;
+        $this->flashMessage = $flashMessage;
+    }
+
+    /**
+     * Lists the walls
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function indexAction()
+    {
+        $walls = $this->em
+            ->getRepository('Integrated\Bundle\SubscriptionBundle\Model\SubscriptionWall')
+            ->findAll();
+
+        return $this->templating->renderResponse('IntegratedSubscriptionBundle:SubscriptionWall:index.html.twig', ['walls' => $walls]);
+    }
+
+    /**
+     * Creates a wall
+     *
+     * @param Request $request
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse|\Symfony\Component\HttpFoundation\Response
+     */
+    public function createAction()
+    {
+        $wall = new SubscriptionWall();
+        $channels = $this->cm->findAll();
+        $channelNames = [];
+        
+        foreach ($channels as $channel) {
+            $channelNames[] = $channel->getName();
+        }
+
+        $form = $this->createCreateForm($wall, $channelNames);
+        $form->handleRequest($this->request);
+
+        if ($form->isValid()) {
+            $this->em->persist($wall);
+            $this->em->flush();
+
+            if (isset($this->request->request->get("integrated_subscription_wall")["channel"])) {
+                foreach ($this->request->request->get("integrated_subscription_wall")["channel"] as $channel) {
+                    $wallChannel = new WallChannel();
+                    $wallChannel->setChannel($channelNames[$channel]);
+                    $wallChannel->setWall($wall);
+                    $wallChannel->setSubscriptionWall($wall);
+                    $this->em->persist($wallChannel);
+                    $this->em->flush();
+                }
+            }
+
+            $this->flashMessage->success('Wall created');
+
+            return new RedirectResponse($this->router->generate("integrated_subscription_show_wall"));
+        }
+        $build['form'] = $form->createView();
```

Comment:
It's nicer not to use a $build variable, but to just build the array inside the function, like you did in the indexaction.